include/phasar/PhasarLLVM/DataFlowSolver/Mono/Solver/MonoCache.h

```diff
@@ -0,0 +1,64 @@
+/******************************************************************************
+ * Copyright (c) 2020 Philipp Schubert.
+ * All rights reserved. This program and the accompanying materials are made
+ * available under the terms of LICENSE.txt.
+ *
+ * Contributors:
+ *     Ajay Subramanya Kudli Prasanna Kumar, Philipp Schubert and others
+ *****************************************************************************/
+
+#ifndef PHASAR_PHASARLLVM_MONO_SOLVER_MONOCACHE_H_
+#define PHASAR_PHASARLLVM_MONO_SOLVER_MONOCACHE_H_
+
+#include<iostream>
+#include<unordered_map>
+#include<vector>
+
+namespace psr {
+  template <class K, class V>
+class MonoCache {
+  // TODO
+public:
+        unordered_map<K, V> summaries;
+        vector<int> time; //The most recently added or recently used key is placed last
```

Comment:
I'm not sure how that mechanism is supposed to work. Could you perhaps add some thorough comments to that entire class?

Reply:
Added thorough comments to the respective class along with some minor changes in the code